Support moving the AI Chat view into a separate window

cdamus opened the pull request:
#### What it does

Adds a toolbar action to move the AI Chat view into a separate window, thusly:

![demo](https://github.com/user-attachments/assets/0a23da96-961a-403a-938f-4c7afcef2a89)

Fixes eclipsesource/osweek-2024#69.

#### How to test

Open the AI Chat view. Press the button to move it into a separate window. Enjoy the freedom of unlimited screen real estate!

#### Follow-ups

- consider whether it might be useful to reopen the view in the main window when the secondary window is closed
- or, alternatively, offer a toolbar action that (when the view is in the secondary window) moves it back to the main window

#### Review checklist

- [X] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

martin-fleck-at commented on packages/ai-chat-ui/src/browser/chat-view-widget.tsx:
```diff
@@ -28,11 +28,13 @@ export namespace ChatViewWidget {
     public static ID = 'chat-view-widget';
     static LABEL = nls.localizeByDefault('Chat');
 
+    public secondaryWindow: Window | undefined;
```
nit: `public` is not necessary to specify.

cdamus replied:
Addressed by commit 93e12398b0df65be3917bef31ca74d01bee7cc6d

martin-fleck-at commented on packages/ai-chat-ui/src/browser/ai-chat-command-contribution.ts:
```diff
@@ -46,8 +57,29 @@ export class AIChatCommandContribution implements CommandContribution {
+    }
+
+    get chatView(): ChatViewWidget | undefined {
+        return this.widgetManager.getWidgets(ChatViewWidget.ID).find(w => w instanceof ChatViewWidget) as ChatViewWidget | undefined;
```
```suggestion
        return this.widgetManager.tryGetWidget(ChatViewWidget.ID);
```

or alternatively:

```
    @inject(AIChatContribution)
    protected aiChatContribution: AIChatContribution;
[...]
        return this.aiChatContribution.tryGetWidget();
```

cdamus replied:
Obsoleted by commit 93e12398b0df65be3917bef31ca74d01bee7cc6d

martin-fleck-at commented on packages/ai-chat-ui/src/browser/ai-chat-command-contribution.ts:
```diff
@@ -46,8 +57,29 @@ export class AIChatCommandContribution implements CommandContribution {
         });
+        commands.registerCommand(ChatCommands.EXTRACT_CHAT_VIEW, {
+            execute: () => this.extractChatView(),
+            isEnabled: () => this.canExtractChatView,
+            isVisible: () => this.canExtractChatView
```
These conditions are not enough to restrict the toolbar items since it is not checking the the argument for `widget`. Please see `withWidget` conditions for other commands, e.g., `ChatCommands.LOCK__WIDGET`. You can see the problem that in the text editor, we get these toolbar items/commands as well.

cdamus replied:
Addressed in commit 93e12398b0df65be3917bef31ca74d01bee7cc6d

martin-fleck-at reviewed (commented):
Change looks good to me in general, I do have some minor comments though.

I think in general it would be great if the toolbar would be shown in the secondary window and that there would be a way to simply re-attach the view again in the main window. But this does not have to be done as part of this PR and is more a Theia-specific problem it seems.

cdamus commented:
> I think in general it would be great if the toolbar would be shown in the secondary window and that there would be a way to simply re-attach the view again in the main window. But this does not have to be done as part of this PR and is more a Theia-specific problem it seems.

Thanks! The toolbar is a general secondary-window issue that we're working on as a stretch goal; not necessarily in this PR.

cdamus commented:
Commit 93e12398b0df65be3917bef31ca74d01bee7cc6d addresses review comments and additionally restores the non-singleton binding of the chat view widget from earlier because it just wasn't working as reworked earlier in this PR on account of the a disposed widget being resurrected. So now it's not bound as a singleton but the factory maintains its own singleton.

martin-fleck-at reviewed (approved):
Changes look good to me in general, thank you very much! This is really a great feature!

I noticed that once you close and re-open the chat (independent from the detachment), the enablement from the auto-scroll locking becomes weirdly unstable. But this is not something that needs to be part of this PR but it should probably be logged.
